In [1]:
import glob
import os
import numpy as np 

## 0) loading the l1c files

In [2]:
#path = '/home/datawork-cersat-public/cache/project/sarwave/data/products/tests/iw/slc/l1c/4.1/'
#safe = 'S1A_IW_XSP__1SDV*.SAFE'
path = '/home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/'
safe = '*IW_XSP__1SDV*B07*.SAFE'


z = glob.glob(path+'*/*/'+safe)
fns = glob.glob(path+'*/*/'+safe+'/*b07*.nc')  # the iw2 sub_swaths have a 5 tile_sample whereas the 2 others (iw1 et iw3) have a 4 tile_sample
print('number of SAFE files : '+str(len(z)), '; number of nc files : '+str(len(fns))) 

fns_array = np.array(fns)
# np.savetxt('/home1/datawork/ljessel/l1c_conversion/4.1/ALL_S1A_IW_XSP__1SDV.txt', fns_array, fmt='%s')
# np.savetxt('/home1/datawork/ljessel/l1c_conversion/ALL_S1AB_IW_XSP__1SDV_B07.txt', fns_array, fmt='%s')
fns_array

number of SAFE files : 33348 ; number of nc files : 99857


array(['/home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/2018/297/S1A_IW_XSP__1SDV_20181024T010502_20181024T010530_024271_02A7FA_FEA1_B07.SAFE/l1c-s1a-iw3-xsp-1sdv-20181024t010504-20181024t010529-024271-02a7fa-b07.nc',
       '/home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/2018/297/S1A_IW_XSP__1SDV_20181024T010502_20181024T010530_024271_02A7FA_FEA1_B07.SAFE/l1c-s1a-iw1-xsp-1sdv-20181024t010502-20181024t010530-024271-02a7fa-b07.nc',
       '/home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/2018/297/S1A_IW_XSP__1SDV_20181024T010502_20181024T010530_024271_02A7FA_FEA1_B07.SAFE/l1c-s1a-iw2-xsp-1sdv-20181024t010503-20181024t010528-024271-02a7fa-b07.nc',
       ...,
       '/home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/2021/044/S1B_IW_XSP__1SDV_20210213T185012_20210213T185038_025592_030CBA_2E42_B07.SAFE/l1c-s1b-iw2-xsp-1sdv-20210213t185012-20210213t185037-025592-030cba-b07.nc',


9007 SAFE et 27_042 fichiers .nc du type :  /home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/* IW_XSP__1SDV* B07*.SAFE/*.nc

9007 SAFE et 26_940 fichiers .nc du type :  /home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/* IW_XSP__1SDV* B07*.SAFE/* b07*.nc

 18 sept 2024 B07 : number of SAFE files : 33348 ; number of nc files : 99857

# Add a filter to select only SAFE from extension [ optional ]

In [3]:
import pandas as pd
# listing_safe_l1B = '/home/datawork-cersat-public/cache/project/sarwave/data/listings/iw_l1b_safe_a13_from_creodias_v14sept24.txt' # extension dataset
listing_safe_l1B = '/home/datawork-cersat-public/project/sarwave/data/listings/iw_l1b_A13_medium_listing.txt' # base listing medium

dfl1b = pd.read_csv(listing_safe_l1B,names=['safe'])
dfl1b

safe
0     /home/datawork-cersat-public/project/sarwave/d...
1     /home/datawork-cersat-public/project/sarwave/d...
2     /home/datawork-cersat-public/project/sarwave/d...
3     /home/datawork-cersat-public/project/sarwave/d...
4     /home/datawork-cersat-public/project/sarwave/d...
...                                                 ...
9196  /home/datawork-cersat-public/project/sarwave/d...
9197  /home/datawork-cersat-public/project/sarwave/d...
9198  /home/datawork-cersat-public/project/sarwave/d...
9199  /home/datawork-cersat-public/project/sarwave/d...
9200  /home/datawork-cersat-public/project/sarwave/d...

[9201 rows x 1 columns]

In [4]:
import os
dfl1c = pd.DataFrame({'safe':[os.path.basename(uu).replace('A13','B07') for uu in dfl1b['safe']]})
dfl1c.values

array([['S1A_IW_XSP__1SDV_20181024T010502_20181024T010530_024271_02A7FA_FEA1_B07.SAFE'],
       ['S1A_IW_XSP__1SDV_20181024T010552_20181024T010619_024271_02A7FA_A536_B07.SAFE'],
       ['S1A_IW_XSP__1SDV_20181024T010527_20181024T010554_024271_02A7FA_333A_B07.SAFE'],
       ...,
       ['S1B_IW_XSP__1SDV_20210213T153523_20210213T153550_025590_030CA9_B725_B07.SAFE'],
       ['S1A_IW_XSP__1SDV_20210213T193912_20210213T193940_036576_044BD7_A9B5_B07.SAFE'],
       ['S1B_IW_XSP__1SDV_20210213T185036_20210213T185103_025592_030CBA_FEDF_B07.SAFE']],
      dtype=object)

In [5]:
for vv in dfl1c['safe'].values:
    if vv == 'S1A_IW_XSP__1SDV_20181024T010502_20181024T010530_024271_02A7FA_FEA1_B07.SAFE':
        print('bingo')

bingo


In [6]:
from collections import defaultdict
from tqdm import tqdm
cpt = defaultdict(int)
consolidated_list_measu = []
safe_kept = []
for ii in tqdm(range(len(fns_array))):
    ff = fns_array[ii]
    basesafe_x  = os.path.basename(os.path.dirname(ff))
    # print('basesafe_x',basesafe_x)
    
    if basesafe_x in dfl1c['safe'].values.squeeze():
        cpt['keep'] += 1   
        if basesafe_x not in safe_kept:
            safe_kept.append(safe_kept)
            cpt['safe_keep'] += 1
        # print('ok')
        consolidated_list_measu.append(ff)
        # dsùfkllsmdkf
    else:
        # print('ko')
        cpt['rejected'] += 1
    # if ii>10:
        # sdmlfkdmslkf
print('cpt',cpt)
consolidated_list_measu = np.array(consolidated_list_measu)

100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 99857/99857 [00:16<00:00, 5924.29it/s]

cpt defaultdict(<class 'int'>, {'keep': 26928, 'safe_keep': 26928, 'rejected': 72929})


In [7]:
# extension cpt defaultdict(<class 'int'>,        {'rejected': 26997, 'keep': 72860, 'safe_keep': 72860})
# medium listing : cpt defaultdict(<class 'int'>, {'rejected': 72929,'keep': 26928, 'safe_keep': 26928}

In [10]:
# if False:
#     fout = '/home1/scratch/agrouaze/l1c_conversion/S1AB_IW_L1C_XSP__1SDV_B07_extension_from_creodias_nc_fullpath.txt'
#     os.makedirs(os.path.dirname(fout),exist_ok=True)
#     np.savetxt(fout, consolidated_list_measu, fmt='%s')
#     print(fout)
# else:
#     print(len(fns_array))
#     fout = '/home1/scratch/agrouaze/l1c_conversion/S1AB_IW_L1C_XSP__1SDV_B07_medium_plus_extension_nc_fullpath.txt'
#     os.makedirs(os.path.dirname(fout),exist_ok=True)
#     np.savetxt(fout, fns_array, fmt='%s')
# print(fout)
print(len(fns_array))
fout = '/home1/scratch/agrouaze/l1c_conversion/S1AB_IW_L1C_XSP__1SDV_B07_medium_listing_nc_fullpath.txt'
os.makedirs(os.path.dirname(fout),exist_ok=True)
np.savetxt(fout, fns_array, fmt='%s')
print(fout)

99857
/home1/scratch/agrouaze/l1c_conversion/S1AB_IW_L1C_XSP__1SDV_B07_medium_listing_nc_fullpath.txt
